File: src/types/document.ts
export const DOC_CATEGORIES = ['Plans & Drawings', 'Contracts', 'Permits', 'Reports', 'Specifications', 'Other'] as const
export type DocCategory = typeof DOC_CATEGORIES[number]

export interface ProjectDocument {
  id: string
  companyId: string
  projectId: string
  name: string
  fileUrl: string
  fileType: string
  fileSize: number
  category: DocCategory
  uploadedById: string
  uploadedByName: string
  createdAt: string
}

export interface CreateDocumentData {
  name: string
  fileUrl: string
  fileType: string
  fileSize: number
  category: DocCategory
}

export interface UploadResult {
  url: string
  fileType: string
  fileSize: number
  name: string
}

File: backend/app/schemas/document.py
import uuid
from datetime import datetime
from pydantic import BaseModel

DOC_CATEGORIES = ['Plans & Drawings', 'Contracts', 'Permits', 'Reports', 'Specifications', 'Other']


class DocumentCreate(BaseModel):
    name: str
    file_url: str
    file_type: str
    file_size: int
    category: str = 'Other'


class DocumentOut(BaseModel):
    id: uuid.UUID
    company_id: uuid.UUID
    project_id: uuid.UUID
    name: str
    file_url: str
    file_type: str
    file_size: int
    category: str
    uploaded_by_id: uuid.UUID
    uploaded_by_name: str
    created_at: datetime
